<a href="https://colab.research.google.com/github/NontobekoNcube/NontobekoN/blob/main/SalesAnalysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

DATA ANALYSIS OF SALES DATA AT

In [2]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import statistics as stats

## Load and Inspect data

In [3]:
from google.colab import files
uploaded = files.upload()
df = pd.read_csv('Sales.csv')


Saving Sales.csv to Sales.csv


## Understanding the data
This section explains the overall structure of the dataset

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7560 entries, 0 to 7559
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   Date    7560 non-null   object
 1   Time    7560 non-null   object
 2   State   7560 non-null   object
 3   Group   7560 non-null   object
 4   Unit    7560 non-null   int64 
 5   Sales   7560 non-null   int64 
dtypes: int64(2), object(4)
memory usage: 354.5+ KB


* The dataset has 7560 records(rows), and 6 colums(features).
* The columns include: Date, Time, State, Group, Unit, Sales.
* The data types in the dataset are integers and objects(strings).
* Nominal variables: Date,State, Group
* Ordinal variables: Time
* Discrete variables: Unit
* Continuous variables: Sales



In [31]:
df.head()


,Date,Time,State,Group,Unit,Sales
0,1-Oct-2020,Morning,WA,Kids,8,20000
1,1-Oct-2020,Morning,WA,Men,8,20000
2,1-Oct-2020,Morning,WA,Women,4,10000
3,1-Oct-2020,Morning,WA,Seniors,15,37500
4,1-Oct-2020,Afternoon,WA,Kids,3,7500


## Handle missing values
There are no null values

## Task 1: Identifying top performing states

In [12]:
unique_states = df["State"].nunique()
print(f"Number of unique states: {unique_states}")

Number of unique states: 7


In [13]:
# Group by state and sum the sales
state_revenue = df.groupby('State')['Sales'].sum()
print(state_revenue)

State
NSW     74970000
NT      22580000
QLD     33417500
SA      58857500
TAS     22760000
VIC    105565000
WA      22152500
Name: Sales, dtype: int64


In [14]:
print(f"Average sales amount: ${df['Sales'].mean():,.2f}")
print(f"Median units sold: {df['Unit'].median()}")
print(f"Most common state: {df['State'].mode()[0]}")  # [0] gets first result
print(f"Most popular demographic: {df['Group'].mode()[0]}")



Average sales amount: $45,013.56
Median units sold: 14.0
Most common state:  NSW
Most popular demographic:  Kids


In [15]:
# Calculate quartiles for the state revenues
import numpy as np

# Get the revenue values
revenues = state_revenue.values

# Calculate quartiles
q1 = np.percentile(revenues, 25)
q3 = np.percentile(revenues, 75)

print(f"1st Quartile (Q1): ${q1:,.0f}")
print(f"3rd Quartile (Q3): ${q3:,.0f}")

# Classify states by performance
print("\n--- STATE PERFORMANCE CLASSIFICATION ---")
print("🔴 BOTTOM PERFORMERS (1st Quartile - Need Sales Programs):")
for state, revenue in state_revenue.items():
    if revenue <= q1:
        print(f"   {state}: ${revenue:,.0f}")

print("\n🟢 TOP PERFORMERS (4th Quartile - Success Stories):")
for state, revenue in state_revenue.items():
    if revenue >= q3:
        print(f"   {state}: ${revenue:,.0f}")

print("\n🟡 MIDDLE PERFORMERS (2nd & 3rd Quartiles):")
for state, revenue in state_revenue.items():
    if q1 < revenue < q3:
        print(f"   {state}: ${revenue:,.0f}")

1st Quartile (Q1): $22,670,000
3rd Quartile (Q3): $66,913,750

--- STATE PERFORMANCE CLASSIFICATION ---
🔴 BOTTOM PERFORMERS (1st Quartile - Need Sales Programs):
    NT: $22,580,000
    WA: $22,152,500

🟢 TOP PERFORMERS (4th Quartile - Success Stories):
    NSW: $74,970,000
    VIC: $105,565,000

🟡 MIDDLE PERFORMERS (2nd & 3rd Quartiles):
    QLD: $33,417,500
    SA: $58,857,500
    TAS: $22,760,000


## Task 2: Develop Sales programs for lower revenue states

Sales analysis of top perfomers

In [20]:
state_revenue = df.groupby('State')['Sales'].sum()
print("State revenues:")
print(state_revenue)

State revenues:
State
NSW     74970000
NT      22580000
QLD     33417500
SA      58857500
TAS     22760000
VIC    105565000
WA      22152500
Name: Sales, dtype: int64


In [39]:
top_data = df[df['State'].isin([' NSW', ' VIC'])]
time_revenue = top_data.groupby('Time')['Sales'].sum().sort_values(ascending=False)
print("Revenue by Time of Day:")
print(time_revenue)

# Convert to percentages
total_revenue = time_revenue.sum()
print("\nAs Percentages:")
for time_period, revenue in time_revenue.items():
    percentage = (revenue / total_revenue) * 100
    print(f"{time_period}: {percentage:.1f}% of revenue")
print("\n CUSTOMER GROUP ANALYSIS FOR TOP PERFORMERS")
group_revenue = top_data.groupby('Group')['Sales'].sum().sort_values(ascending=False)
print("Revenue by Customer Group:")
print(group_revenue)

# Convert to percentages
total_revenue_groups = group_revenue.sum()
print("\nAs Percentages:")
for group, revenue in group_revenue.items():
    percentage = (revenue / total_revenue_groups) * 100
    print(f"{group}: {percentage:.1f}% of revenue")

Revenue by Time of Day:
Time
Morning      60770000
Afternoon    60172500
Evening      59592500
Name: Sales, dtype: int64

As Percentages:
 Morning: 33.7% of revenue
 Afternoon: 33.3% of revenue
 Evening: 33.0% of revenue

 CUSTOMER GROUP ANALYSIS FOR TOP PERFORMERS
Revenue by Customer Group:
Group
Women      45655000
Men        45430000
Kids       44947500
Seniors    44502500
Name: Sales, dtype: int64

As Percentages:
 Women: 25.3% of revenue
 Men: 25.2% of revenue
 Kids: 24.9% of revenue
 Seniors: 24.7% of revenue


Sales analysis of bottom perfomers

In [29]:
# Analyze NT and WA patterns
bottom_data = df[df['State'].isin([' NT', ' WA'])]

print("BOTTOM PERFORMERS (NT + WA) ANALYSIS")
print(f"Total bottom performer revenue: ${bottom_data['Sales'].sum():,}")

print("\n TIME ANALYSIS FOR BOTTOM PERFORMERS")
time_revenue_bottom = bottom_data.groupby('Time')['Sales'].sum().sort_values(ascending=False)
print("Revenue by Time of Day:")
print(time_revenue_bottom)

# Convert to percentages
total_revenue_bottom = time_revenue_bottom.sum()
print("\nAs Percentages:")
for time_period, revenue in time_revenue_bottom.items():
    percentage = (revenue / total_revenue_bottom) * 100
    print(f"{time_period}: {percentage:.1f}% of revenue")

print("\n CUSTOMER GROUP ANALYSIS FOR BOTTOM PERFORMERS")
group_revenue_bottom = bottom_data.groupby('Group')['Sales'].sum().sort_values(ascending=False)
print("Revenue by Customer Group:")
print(group_revenue_bottom)

# Convert to percentages
total_revenue_groups_bottom = group_revenue_bottom.sum()
print("\nAs Percentages:")
for group, revenue in group_revenue_bottom.items():
    percentage = (revenue / total_revenue_groups_bottom) * 100
    print(f"{group}: {percentage:.1f}% of revenue")

BOTTOM PERFORMERS (NT + WA) ANALYSIS
Total bottom performer revenue: $44,732,500

 TIME ANALYSIS FOR BOTTOM PERFORMERS
Revenue by Time of Day:
Time
Morning      15010000
Evening      14930000
Afternoon    14792500
Name: Sales, dtype: int64

As Percentages:
 Morning: 33.6% of revenue
 Evening: 33.4% of revenue
 Afternoon: 33.1% of revenue

 CUSTOMER GROUP ANALYSIS FOR BOTTOM PERFORMERS
Revenue by Customer Group:
Group
Men        11515000
Kids       11325000
Seniors    10977500
Women      10915000
Name: Sales, dtype: int64

As Percentages:
 Men: 25.7% of revenue
 Kids: 25.3% of revenue
 Seniors: 24.5% of revenue
 Women: 24.4% of revenue


In [30]:
# Average sales per transaction
top_avg = df[df['State'].isin([' NSW', ' VIC'])]['Sales'].mean()
bottom_avg = df[df['State'].isin([' NT', ' WA'])]['Sales'].mean()
print(f"Top performers avg transaction: ${top_avg:,.0f}")
print(f"Bottom performers avg transaction: ${bottom_avg:,.0f}")

Top performers avg transaction: $83,581
Bottom performers avg transaction: $20,709


Top and Bottom performers show a balanced distribution of sales in terms of times when sales are done as well as the demographics that buy. The issue is then not that the bottom have a specialized approach for instance, selling at a specific time of day, to a specific demographic.  

Further Analysis to determine what is driving higher sales for top performing states:

In [32]:
# Check average units per transaction
top_units = df[df['State'].isin([' NSW', ' VIC'])]['Unit'].mean()
bottom_units = df[df['State'].isin([' NT', ' WA'])]['Unit'].mean()

print(f"Top performers avg units per transaction: {top_units:.1f}")
print(f"Bottom performers avg units per transaction: {bottom_units:.1f}")

# Check average price per unit
top_price_per_unit = df[df['State'].isin([' NSW', ' VIC'])]['Sales'].sum() / df[df['State'].isin([' NSW', ' VIC'])]['Unit'].sum()
bottom_price_per_unit = df[df['State'].isin([' NT', ' WA'])]['Sales'].sum() / df[df['State'].isin([' NT', ' WA'])]['Unit'].sum()

print(f"\nTop performers avg price per unit: ${top_price_per_unit:,.0f}")
print(f"Bottom performers avg price per unit: ${bottom_price_per_unit:,.0f}")

Top performers avg units per transaction: 33.4
Bottom performers avg units per transaction: 8.3

Top performers avg price per unit: $2,500
Bottom performers avg price per unit: $2,500


### Key Findings and Recommendations

Same price per unit: $2,500 everywhere
But top performers sell 4x more units per transaction: 33.4 vs 8.3 units
This means the revenue gap is purely about VOLUME per sale, not pricing.

1. Volume Discount Programs: "Buy 20+ units, get 10% off"
2. Bulk Sales Training: Train sales teams to suggest larger quantities
3. Inventory Management: Ensure larger volumes are available for bigger orders
4. Business Customer Focus: Target commercial buyers who need larger quantities
5. Cross-selling: "If you need 8 units of Product A, you probably need Product B too"
6. Order Minimum Incentives: "Free shipping on orders of 25+ units"